### PR TITLE
ARM64: Moves RA functions to header

### PR DESCRIPTION
**Pull request by Sonicadvance1**
These are just some basic address calculations and a load, we want these to be inlined as much as possible.

### CODE REVIEW

File: External/FEXCore/Source/Interface/Core/JIT/Arm64/JITClass.h [hunk 1]
```diff
@@ -89,25 +89,89 @@ class Arm64JITCore final : public CPUBackend, public Arm64Emitter  {
...
   [[nodiscard]] aarch64::Register GetReg(IR::NodeID Node) const;
 
+  template<uint8_t RAType>
+  [[nodiscard]] std::pair<aarch64::Register, aarch64::Register> GetSrcPair(IR::NodeID Node) const;
+
+  [[nodiscard]] FEXCore::IR::RegisterClassType GetRegClass(IR::NodeID Node) const;
+
   template<>
-  [[nodiscard]] aarch64::Register GetReg<RA_32>(IR::NodeID Node) const;
+  [[nodiscard]] aarch64::Register GetReg<Arm64JITCore::RA_32>(IR::NodeID Node) const {
```
Comment (lioncash) [inline External/FEXCore/Source/Interface/Core/JIT/Arm64/JITClass.h:98]: Seems like this version can call the RA_64 version and call `.W()` on the result. Not sure if that would be desirable though

Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/JIT/Arm64/JITClass.h:98]: Could definitely do that. We are going to be paying the vixl cost of this `.W()` ending up in the stack regardless.

Comment (lioncash) [reply External/FEXCore/Source/Interface/Core/JIT/Arm64/JITClass.h:98]: yeah, was more just considering unifying the underlying behavior into one spot. Though either way is fine, considering how small it is

Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/JIT/Arm64/JITClass.h:98]: Resolved.

**Review by lioncash — APPROVED** [review]

**Pull request merged**